openapi/provider_test.go

```diff
@@ -193,6 +195,134 @@ definitions:
 	})
 }
 
+func Test_colliding_resource_names(t *testing.T) {
+	makeSwaggerDoc := func(path1, preferredName1, path2, preferredName2 string) string {
+		if path1 == "" {
+			path1 = "/v1/abc"
+		}
+		if path2 == "" {
+			path2 = "/v1/xyz"
+		}
+		xTerraformResourceName1 := ""
+		if preferredName1 != "" {
+			xTerraformResourceName1 = `x-terraform-resource-name: "` + preferredName1 + `"`
+		}
+		xTerraformResourceName2 := ""
+		if preferredName2 != "" {
+			xTerraformResourceName2 = `x-terraform-resource-name: "` + preferredName2 + `"`
+		}
+		swagger := `swagger: "2.0"
+paths:
+  ` + path1 + `:
+    post:
+      ` + xTerraformResourceName1 + `
+      parameters:
+      - in: "body"
+        name: "body"
+        schema:
+          $ref: "#/definitions/whatever"
+      responses:
+        201:
+          schema:
+            $ref: "#/definitions/whatever"
+  ` + path1 + `/{id}:
+    get:
+      parameters:
+      - name: "id"
+        in: "path"
+        type: "string"
+      responses:
+        200:
+          schema:
+            $ref: "#/definitions/whatever"
+
+  ` + path2 + `:
+    post:
+      ` + xTerraformResourceName2 + `
+      parameters:
+      - in: "body"
+        name: "body"
+        schema:
+          $ref: "#/definitions/whatever"
+      responses:
+        201:
+          schema:
+            $ref: "#/definitions/whatever"
+  ` + path2 + `/{id}:
+    get:
+      parameters:
+      - name: "id"
+        in: "path"
+        type: "string"
+      responses:
+        200:
+          schema:
+            $ref: "#/definitions/whatever"
+
+definitions:
+  whatever:
+    type: "object"
+    properties:
+      id:
+        type: "string"
+        readOnly: true`
+		return swagger
+	}
+
+	Convey("Given a swagger doc that declares resources with colliding names, "+
```

Comment:
+1 to this convey style
less verbose and more readable: however at this point, won't be better to have this comment' style in a t.Run() ?

Reply:
I agree.  I'll make the change.